mlir/lib/Dialect/Affine/Transforms/AffineLoopInterchange.cpp

```diff
@@ -0,0 +1,761 @@
+    funcOp.walk([&](AffineForOp op) {
+      loopNest.push_back(op);
+      if (op.getParentOp()->getName().getStringRef() == "affine.for")
+        forTree[op.getOperation()->getParentOp()].push_back(op);
+
+      forOperations[op.getOperation()] = op;
+    });
+    // Separate one of the sibling at a time.
+    for (auto &loopNest : forTree) {
+      // This loop nest has no siblings problem. Check the next loop nest.
+      if (loopNest.second.size() < 2)
+        continue;
+      oneChild = false;
+      separateSiblingLoops(forOperations[loopNest.first],
+                           loopNest.second.back(), loopNest.second);
+      // We need to walk the function again since the structure of loop nests
+      // within the funcOp body has changed.
+      break;
+    }
+    loopNest.clear();
+    forTree.clear();
+    forOperations.clear();
+  }
+  return;
+}
+
+/// Scans the loop nest rooted at `rootForOp` and collects all affine.load and
+/// affine.store ops. Fills `loadAndStoreOps` with all such ops.
+static void getAllLoadStores(AffineForOp rootForOp,
+                             SmallVector<Operation *, 8> &loadAndStoreOps) {
+  rootForOp.getOperation()->walk([&](Operation *op) {
+    if (isa<AffineLoadOp>(op) || isa<AffineStoreOp>(op)) {
+      loadAndStoreOps.push_back(op);
+    }
+  });
+}
+
+/// Fills `elementsSize` with the size of element types of respective memrefs
+/// accessed by the ops in `loadAndStoreOps`. These will be later used to
+/// check if two accesses are within a cache_line_size/element_size distance
+/// apart for a useful locality.
+static void getElementsSize(SmallVector<Operation *, 8> &loadAndStoreOps,
+                            DenseMap<Operation *, unsigned> &elementsSize) {
+  // Assumption: In cases where element size is difficult to obtain, assume a
+  // default value of 8 bytes.
+  constexpr unsigned defaultEltSize = 8;
+  MemRefType memrefType;
+  for (Operation *op : loadAndStoreOps) {
+    if (isa<AffineLoadOp>(op)) {
+      AffineLoadOp loadOp = cast<AffineLoadOp>(*op);
+      memrefType = loadOp.getMemRefType();
+    } else if (isa<AffineStoreOp>(op)) {
+      AffineStoreOp storeOp = cast<AffineStoreOp>(*op);
+      memrefType = storeOp.getMemRefType();
+    }
+    elementsSize[op] = memrefType.hasStaticShape()
+                           ? getMemRefSizeInBytes(memrefType).getValue() /
+                                 memrefType.getNumElements()
+                           : defaultEltSize;
+  }
+}
+
+/// Calculates the loop-carried-dependence vector for this loop nest rooted at
+/// `rootForOp`. A value `true` at i-th index means that loop at depth i in the
+/// loop nest carries a dependence.
+static void getLoopCarriedDependenceVector(
+    AffineForOp &rootForOp, ArrayRef<Operation *> loadAndStoreOps,
+    unsigned loopNestSize, SmallVector<bool, 4> &loopCarriedDependenceVector) {
+
+  // Resize the `loopCarriedDependenceVector` to fit entire loop nest.
```

Comment:
Rephrase this - you aren't resizing a single vector.

Reply:
Fixed. Thanks.